src/client/rest/mod.rs

```diff
@@ -1182,6 +1182,7 @@ pub fn get_messages(channel_id: u64, query: &str)
                            || client.get(&url))?;
 
     serde_json::from_reader::<HyperResponse, Vec<Message>>(response).map_err(From::from)
+        .map(|mut x| x.iter_mut().map(|x| x.clone().improved_content()).collect::<Vec<Message>>())
```

Comment:
This clone feels unnecessary. You should be able to change the `iter_mut` into an `into_iter` to remove the clone.